### PR TITLE
Gallery: update NEMO example

### PR DESCRIPTION
**Pull request by rcomer**
### Description

* Why use four lines when you can use one?
* It took me a while to understand why the time coordinate promotion was necessary, so expanded that comment.
---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/gallery_code/oceanography/plot_load_nemo.py [hunk 1]
```diff
@@ -13,24 +13,21 @@
...
-            cube.attributes[attribute] = ""
+    # Some attributes are unique to each file and must be removed to allow
+    # concatenation.
+    equalise_attributes(cubes)
 
     # The cubes still cannot be concatenated because their time dimension is
-    # time_counter rather than time. time needs to be promoted to allow
-    # concatenation.
+    # time_counter, which has the same value for each cube, rather than time,
+    # which varies. time needs to be promoted to allow concatenation.
```
Comment (trexfeathers) [inline docs/gallery_code/oceanography/plot_load_nemo.py:30]: I agree this was confusing. But I'm still finding it confusing!

Could you rephrase the whole thing to refer explicitly to `DimCoord`s and `AuxCoord`s? And it also hadn't occurred to me that promotion also demotes `time_counter` (perhaps I should have worked that one out); do you think it's worth mentioning that?

**Review by trexfeathers — CHANGES_REQUESTED** [review]

File: docs/gallery_code/oceanography/plot_load_nemo.py (hunk 1 above)
Comment (rcomer) [reply docs/gallery_code/oceanography/plot_load_nemo.py]: Words are hard!  How's this? https://github.com/SciTools/iris/pull/4741/commits/848e53f1fcddae27f48f89fa78bfcc837dadda66

Note that, although the "time_counter" coord becomes an aux_coord on the cube, it is still of type `DimCoord` 🤪

**Comment by rcomer** [comment]
Rebased to fix whatsnew conflict

**Review by trexfeathers — APPROVED** [review]

**Comment by trexfeathers** [comment]
Nice bit of continuous improvement @rcomer 👍
